packages/ckeditor5-dev-release-tools/lib/release/updateversions.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+const { globSync } = require( 'glob' );
+const fs = require( 'fs-extra' );
+const semver = require( 'semver' );
+const { normalizeTrim, toUnix, dirname } = require( 'upath' );
+const { tools } = require( '@ckeditor/ckeditor5-dev-utils' );
+const getPackageJson = require( '../utils/getpackagejson' );
+
+/**
+ * The purpose of the script is to update the version of a root package found in the current working
+ * directory and packages if the `options.packagesDirectory` path is provided.
+ *
+ * @param {Object} options
+ * @param {String} options.version Version to store in a `package.json` file under the `version` key.
+ * @param {String} [options.packagesDirectory] Relative path to a location of packages to update. If not specified,
+ * only the root package is checked.
+ * @param {String} [options.cwd=process.cwd()] Current working directory from which all paths will be resolved.
+ */
+module.exports = function updateVersions( { packagesDirectory, version, cwd = process.cwd() } ) {
+	const normalizedCwd = toUnix( cwd );
+	const normalizedPackagesDir = packagesDirectory ? normalizeTrim( packagesDirectory ) : null;
+	const globPatterns = [ 'package.json' ];
+
+	if ( packagesDirectory ) {
+		globPatterns.push( normalizedPackagesDir + '/*/package.json' );
+	}
+
+	const pkgJsonPaths = globSync( globPatterns, { cwd: normalizedCwd, absolute: true, nodir: true } );
+	const randomPackagePath = getRandomPackagePath( pkgJsonPaths, normalizedPackagesDir );
+
+	checkVersionGreaterThanOldVersion( version, getPackageJson( normalizedCwd ).version );
+	checkVersionIsAvailableInNpm( version, getPackageJson( randomPackagePath ).name );
+
+	for ( const pkgJsonPath of pkgJsonPaths ) {
+		const pkgJson = fs.readJsonSync( pkgJsonPath );
+
+		pkgJson.version = version;
+		fs.writeJsonSync( pkgJsonPath, pkgJson, { spaces: 2 } );
+	}
+};
+
+/**
+ * @param {Array<String>} pkgJsonPaths
+ * @param {String|null} packagesDirectory
+ * @returns {Object}
+ */
+function getRandomPackagePath( pkgJsonPaths, packagesDirectory ) {
+	const randomPkgJsonPaths = packagesDirectory ?
+		pkgJsonPaths.filter( packagePath => packagePath.includes( packagesDirectory ) ) :
+		pkgJsonPaths;
+	const randomPkgJsonPath = randomPkgJsonPaths[ Math.floor( Math.random() * randomPkgJsonPaths.length ) ];
+
+	return dirname( randomPkgJsonPath );
+}
+
+/**
+ * @param {String} newVersion
+ * @param {String} currentVersion
+ */
+function checkVersionGreaterThanOldVersion( newVersion, currentVersion ) {
+	if ( !semver.valid( newVersion ) ) {
+		throw new Error( `Provided version ${ newVersion } must be a valid semver version.` );
+	}
+
+	if ( !semver.gt( newVersion, currentVersion ) ) {
+		throw new Error( `Provided version ${ newVersion } must be greater than ${ currentVersion }.` );
+	}
+}
+
+/**
```

Comment:
One week later, I am trying to remember the purpose of the npm check. It deserves a comment about why we do it.

This check aims to verify if the proposed version is available. In other words, check if the package can be published under the given version.